taskfiles/rpk.yml

```diff
@@ -6,8 +6,6 @@ tasks:
     vars:
       RPK_VERSION: '{{default "latest" .TAG_NAME}}'
     env:
-      GOOS: '{{OS | lower}}'
```

Comment:
this change seems unrelated to adding plugin support IMO. 

ok to have, but probably makes sense to have a separate PR for this. We'll need @ivotron 's thoughts on this. I think on the build machines you have multiple go packages installed (or we used to)

Reply:
It was something I ran into while adding the `taskfile` mod tidy bit brought up in an earlier comment; I've dropped it from this PR. If possible, I do think this is worth revisiting.

Reply:
we currently don't build for multiple OSs using taskfiles, so we could drop it. We just need to update the places where this is assumed to be read, which is mainly our packaging tools